spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -466,47 +466,50 @@ def _measure_tissue_bridges(self, im_lesion_data, p_lst, idx):
-            # Get the axial slices corresponding to the minimum bridge widths
-            min_dorsal_bridge_width_slice = df_temp.loc[df_temp['dorsal_bridge_width'].idxmin(), 'axial_slice']
-            min_ventral_bridge_width_slice = df_temp.loc[df_temp['ventral_bridge_width'].idxmin(), 'axial_slice']
-
-            # Add a new column with value True to tissue_bridges_df; this information is needed for plotting
-            tissue_bridges_df.loc[(tissue_bridges_df['sagittal_slice'] == sagittal_slice) &
-                                  (tissue_bridges_df['axial_slice'] ==
-                                   min_dorsal_bridge_width_slice), 'min_dorsal_bridge_axial_slice'] = True
-            tissue_bridges_df.loc[(tissue_bridges_df['sagittal_slice'] == sagittal_slice) &
-                                  (tissue_bridges_df['axial_slice'] ==
-                                   min_ventral_bridge_width_slice), 'min_ventral_bridge_axial_slice'] = True
-            # Replace NaN with False
-            tissue_bridges_df.fillna(False, inplace=True)
-
-            # Get the width of the tissue bridges in mm
+
+            # Get the width of the tissue bridges in mm (by multiplying by p_lst[1]) and use np.cos(self.angles[SLICE])
+            # to correct for the angle of the spinal cord with respect to the axial slice
             # NOTE: the orientation is RPI (because we reoriented the image to RPI using orient2rpi()); therefore
             # p_lst[0] is the pixel size in the R-L direction, p_lst[1] is the pixel size in the A-P direction, and
             # p_lst[2] is the pixel size in the S-I direction.
             # Since we are computing dorsal and ventral tissue bridges, we use p_lst[1] (A-P direction)
-            # NOTE: we use np.cos(self.angles[SLICE]) to correct for the angle of the spinal cord with respect to the
-            # slice
-            dorsal_bridge_width_mm = (float(df_temp[df_temp['axial_slice'] == min_dorsal_bridge_width_slice]
-                                            ['dorsal_bridge_width'] * p_lst[1]) *
-                                      np.cos(self.angles[min_dorsal_bridge_width_slice]))
-            ventral_bridge_width_mm = (float(df_temp[df_temp['axial_slice'] == min_ventral_bridge_width_slice]
-                                             ['ventral_bridge_width'] * p_lst[1]) *
-                                       np.cos(self.angles[min_ventral_bridge_width_slice]))
-            total_bridge_width_mm = dorsal_bridge_width_mm + ventral_bridge_width_mm
+            dorsal_bridge_width_mm = df_temp.apply(lambda row: row['dorsal_bridge_width'] * p_lst[1] *
+                                                               np.cos(self.angles[row['axial_slice']]), axis=1)
+            ventral_bridge_width_mm = df_temp.apply(lambda row: row['ventral_bridge_width'] * p_lst[1] *
+                                                                np.cos(self.angles[row['axial_slice']]), axis=1)
```

Comment:
From in-person discussion: I'm concerned that we may be using the wrong angle here for this purpose. `self.angles[i]` is the 3D angle between vertical (the z-direction) and the spinal cord centerline. But here, we want to be using the 2D angle in the sagittal plane. (The difference would occur mostly when there is some scoliosis, I think.)

I'll suggest a fix for this in an extra commit.

Reply:
Thank you for the quick fix! Fortunately, the impact of this is fix is very subtle (second decimal and only for some slices -- see cursor):

![Kapture 2024-07-25 at 14 27 23](https://github.com/user-attachments/assets/fcfd3f76-b190-47b6-91a6-2b746a9cd490)

Reply:
Nice, thanks for this subtle fix, Mathieu! ❤️